File: Compiladores(ano passado)/compact12b/compact/java/NodeUnary.java
import java.util.*;

public class NodeUnary extends Node {
	protected Node arg;
	NodeUnary(Node arg1) { super(); arg = arg1; }
	NodeUnary(Node arg1, int lineno) { super(lineno); arg = arg1; }
	public Node first() { return arg; }
	public void accept(Compiler c) { c.nodeUnary(this); }
}

File: Compiladores(ano passado)/compact12b/compact/c++/NodeUnary.cpp
#include "NodeUnary.hpp"

void NodeUnary::accept(Compiler& c) { c.nodeUnary(*this); }
NodeUnary::~NodeUnary() { delete arg; }
void NodeUnary::print(std::ostream *out) {
	if (out) (*out) << "NodeUnary: ";
	arg->print(out);
	Node::print(out);
}
